pkg/chunk/cassandra/storage_client.go

```diff
@@ -435,7 +484,26 @@ func (s *StorageClient) getChunk(ctx context.Context, decodeContext *chunk.Decod
 	return input, err
 }
 
-func (s *StorageClient) DeleteChunk(ctx context.Context, chunkID string) error {
+func (s *ObjectClient) DeleteChunk(ctx context.Context, chunkID string) error {
 	// ToDo: implement this to support deleting chunks from Cassandra
 	return chunk.ErrMethodNotImplemented
 }
+
+// Stop implement chunk.ObjectClient.
+func (s *ObjectClient) Stop() {
+	s.readSession.Close()
+	s.writeSession.Close()
+}
+
+type noopConvictionPolicy struct{}
+
+// AddFailure should return `true` if the host should be convicted, `false` otherwise.
+// Convicted means connections are removed - we don't want that.
+// Implementats gocql.ConvictionPolicy.
+func (noopConvictionPolicy) AddFailure(err error, host *gocql.HostInfo) bool {
```

Comment:
Should we add a metric here?

Also, what happens if a cassandra node _really_ goes down?  I think we would want gocql to remove it from the pool.  I feel like failure rate would be the best way to determine this, but unsure if that is possible to calculate.

This will also prevent Cortex from ever backing off a struggling Cassandra.  Have we implemented a way for Cassandra to exert backpressure on Cortex?

Reply:
If the host is really failed, connections won't succeed and connection pool will be empty - and the host will be skipped by the query executor.  So I don't think this is a real problem.

As for backing off from a struggling Cassandra, we have timeouts and retries which will pick another host.  Honestly trying to do "clever" load balancing almost always leads to problems like the ones this PR is trying to fix, so I prefer not to do it.